WTC_Libs/stateMachine/comms.c

```diff
@@ -94,9 +94,9 @@ void commsFromGround(uint8_t* buffer) {
 	 */
 	char prompt[100] = { };
 	char promptToGround[100] = { };
-	commsRx->designator = fromBinary((uint8_t*)DMABUFFER[1]);
+	commsRx->designator = fromBinary((uint8_t*)(DMABUFFER + 1));
 
-	sprintf(prompt, "%02x\t%02x",(uint8_t*)DMABUFFER[1], (uint8_t*)commsRx->designator);
```

Comment:
Removed the `uint8_t*` cast, since `sprintf` expects `char*` and complains about signedness otherwise. Additionally, the `%x` formatting specifier expects an integer, not a pointer.